### PR TITLE
support DataType Map

### PR DESCRIPTION
**Pull request by zyklon2**

### CODE REVIEW

**Comment by zyklon2** [comment]
can anyone review this?

File: clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/BatchInsertITest.java [hunk 1]
```diff
@@ -198,4 +204,34 @@ public void successfullyBatchInsertDateTime() throws Exception {
...
 
     }
+    
+    @Test
+    public void successfullyBatchInsertMap() throws Exception {
+        withStatement(statement -> {
+
+            statement.execute("SET allow_experimental_map_type = 1");
+            statement.execute("DROP TABLE IF EXISTS test");
+            statement.execute("CREATE TABLE test(tags Map(String, String))ENGINE=Log");
```
Comment (sundy-li) [inline clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/BatchInsertITest.java:214]: LGTM, Can you add more NestedTypes type tests for these?

1. Add more key/value like 

```
map.put("key" + i, "value" + i);
map.put("key2" + i, "value2" + i);
```

2. Add More types 
```
Map(String, UInt16)
Map(String, Tuple(UInt16, UInt32)
Map(String, Array(UInt16)

```

Comment (zyklon2) [reply clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/BatchInsertITest.java:214]: ok

**Review by sundy-li — APPROVED** [review]
LGTM

**Pull request merged**